src/Orleans.Logging.Legacy/LoggerExtensionMethods.cs

```diff
@@ -23,17 +23,28 @@ public static Logger GetLogger(this Grain grain, string loggerName)
             var loggerFactory = grain.Runtime.ServiceProvider.GetRequiredService<ILoggerFactory>();
             return new LoggerWrapper(loggerName, loggerFactory);
         }
-
+        private const string GrainLoggerKey = "GrainLogger";
```

Comment:
suggest GrainLoggerKey be a static object.

private static object GrainLoggerKey = new object(); 

This ensures uniqueness in bag.  No chance someone else may also try to cache items with the same name because they don't know we've already reserved the name "GrainLogger".